teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -249,10 +249,17 @@ of Catalogs will make regular releases of Catalogs - the release cadence for the
 in [TEP-0079][tep-0079]. The release notes should provide details about the changes in the specific resources in the
 Catalog in a given release, including notes about breaking changes.
 
-The versions in the Artifact Hub should use the simple version of [semver][semver] - `<major-version>.<minor-version>.0`. We propose that the release tags in Catalogs follow the simplified semantic versioning model - `v<major-version>.<minor-version>`. As such, the tags in the Catalogs will map to versions in the Artifact Hub e.g. tag "v0.7" in a Catalog maps to version "0.7.0" in the Hub. Minor versions are made if there are backwards-compatible changes only in the Catalog, while major versions are made if there are any backwards-incompatible changes in the Catalog. The Hub will recognize this versioning model only, therefore Catalogs must use this versioning to surface versions in the Hub.
+The versions in the Artifact Hub should use [semver][semver] - `<major-version>.<minor-version>.<patch-version>` and this value should match the value of the `app.kubernetes.io/version` label in the resource yaml file. We propose that the release tags in Catalogs follow the semantic versioning model - `v<major-version>.<minor-version>.<patch-version>`. As such, the tags in the Catalogs will map to versions in the Artifact Hub, e.g. tag "v0.7.0" in a Catalog maps to version "0.7.0" in the Hub. Patch versions are made for backward-compatible bug fixes, minor versions are made if there are backward-compatible new features in the Catalog, while major versions are made if there are any backward-incompatible changes in the Catalog. 
+
+Specifically, use patch versions when:
+1. Backward-compatible bug fixes for the features supported in the `<major-verion>.<minor-version>`. For example, fixing the authentication bugs to fetch private repositories in the [git-clone task](https://artifacthub.io/packages/tekton-task/quan-git-clone/git-clone).
+2. Backward-compatible security updates in the resource file.
+3. Backward-compatible updates of the container images that are used in the resource.
```

Comment:
Hi @afrittoli, these are great questions. And this is what I'm thinking:

1. If we update the container images for backward-compatible bug fixes/security update for the image itself, we use a patch version for the resource. If we update the images for a new feature, we use a new minor/major version for the resource.
2. If the image is parameterized, we apply the above rule to the default value of the parameter.

In the `CLI` example, I think it should be backward-incompatible since the behaviour of the `task` is changed after changing the container image.

WDYT?